### PR TITLE
More lint and small changes from Blockly

### PR DESCRIPTION
**Pull request by rachel-fenichel**
Incremental progress toward a merge.  I'll add comments to some of the lines that are making actual changes instead of just indentation changes.

### CODE REVIEW

File: core/block.js [hunk 1]
```diff
@@ -1188,11 +1188,12 @@ Blockly.Block.prototype.appendDummyInput = function(opt_name) {
...
  */
 Blockly.Block.prototype.jsonInit = function(json) {
+  var warningPrefix = json['type'] ? 'Block "' + json['type'] + '": ' : '';
 
   // Validate inputs.
-  goog.asserts.assert(json['output'] == undefined ||
-      json['previousStatement'] == undefined,
-      'Must not have both an output and a previousStatement.');
+  goog.asserts.assert(
+      json['output'] == undefined || json['previousStatement'] == undefined,
```
Comment (rachel-fenichel) [inline core/block.js:1195]: Add more information to warning messages when JSON validation fails.

File: core/block.js [hunk 2]
```diff
@@ -1343,11 +1344,11 @@ Blockly.Block.prototype.interpolate_ = function(message, args, lastDummyAlign) {
     var token = tokens[i];
     if (typeof token == 'number') {
       if (token <= 0 || token > args.length) {
-        throw new Error('Block \"' + this.type + '\": ' +
+        throw new Error('Block "' + this.type + '": ' +
```
Comment (rachel-fenichel) [inline core/block.js:1347]: Remove unnecessary escape characters.

File: core/blockly.js [hunk 3]
```diff
@@ -365,19 +365,28 @@ Blockly.jsonInitFactory_ = function(jsonDef) {
  * @param {!Array.<!Object>} jsonArray An array of JSON block definitions.
  */
 Blockly.defineBlocksWithJsonArray = function(jsonArray) {
-  for (var i = 0, elem; elem = jsonArray[i]; i++) {
-    var typename = elem.type;
-    if (typename == null || typename === '') {
-      console.warn('Block definition #' + i +
-        ' in JSON array is missing a type attribute. Skipping.');
+  for (var i = 0; i < jsonArray.length; i++) {
```
Comment (rachel-fenichel) [inline core/blockly.js:368]: JSON array with null or undefined value will now skip the offending item, and proceed to load following items.

File: core/bubble.js [hunk 4]
```diff
@@ -371,11 +369,16 @@ Blockly.Bubble.prototype.registerResizeEvent = function(callback) {
 
 /**
  * Move this bubble to the top of the stack.
+ * @return {!boolean} Whether or not the bubble has been moved.
  * @private
  */
 Blockly.Bubble.prototype.promote_ = function() {
   var svgGroup = this.bubbleGroup_.parentNode;
-  svgGroup.appendChild(this.bubbleGroup_);
+  if (svgGroup.lastChild !== this.bubbleGroup_) {
```
Comment (rachel-fenichel) [inline core/bubble.js:377]: Don't do the appendChild if this is already at the top of the stack (minor performance improvement).

File: core/comment.js [hunk 5]
```diff
@@ -213,17 +213,18 @@ Blockly.Comment.prototype.setVisible = function(visible) {
...
-Blockly.Comment.prototype.textareaFocus_ = function(/*e*/) {
+Blockly.Comment.prototype.textareaFocus_ = function(_e) {
   // Ideally this would be hooked to the focus event for the comment.
   // However doing so in Firefox swallows the cursor for unknown reasons.
   // So this is hooked to mouseup instead.  No big deal.
-  this.bubble_.promote_();
-  // Since the act of moving this node within the DOM causes a loss of focus,
-  // we need to reapply the focus.
-  this.textarea_.focus();
+  if (this.bubble_.promote_()) {
```
Comment (rachel-fenichel) [inline core/comment.js:223]: Don't need to reapply the focus if it didn't move--and promote_ now returns false if nothing happened.

File: core/extensions.js [hunk 6]
```diff
@@ -373,24 +376,24 @@ Blockly.Extensions.buildTooltipForDropdown = function(dropdownName,
...
-    var dropdown = block.getField(dropdownName);
-    if (!dropdown.isOptionListDynamic()) {
-      var options = dropdown.getOptions();
-      for (var i = 0; i < options.length; ++i) {
-        var optionKey = options[i][1]; // label, then value
-        if (lookupTable[optionKey] == null) {
-          console.warn('No tooltip mapping for value ' + optionKey +
-            ' of field ' + dropdownName + ' of block type ' + block.type);
-        }
+Blockly.Extensions.checkDropdownOptionsInTable_ = function(block, dropdownName,
```
Comment (rachel-fenichel) [inline core/extensions.js:382]: Just indentation changes, even though it's on a whole block of code.

File: core/field_variable.js [hunk 7]
```diff
@@ -359,11 +359,10 @@ Blockly.FieldVariable.prototype.onItemSelected = function(menu, menuItem) {
 };
 
 /**
- * Whether this field references any Blockly variables.  If true it may need to
- * be handled differently during serialization and deserialization.  Subclasses
- * may override this.
- * @return {boolean} True if this field has any variable references.
+ * Overrides referencesVariables(), indicating this field refers to a variable.
```
Comment (rachel-fenichel) [inline core/field_variable.js:362]: Changed per review in Blockly.

**Review by thisandagain — APPROVED** [review]
LG

**Review by kchadha — APPROVED** [review]

**Pull request merged**